src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -158,10 +158,20 @@ public override void VisitConstructorDeclaration(ConstructorDeclarationSyntax no
             Visit(node.ExpressionBody, enclosing);
         }
 
+        public override void VisitClassDeclaration(ClassDeclarationSyntax node)
+        {
+            VisitTypeDeclaration(node);
```

Comment:
> What scenarios does this change affect? (Which tests?)

Pretty much any test with base arguments on a class,
```BaseArguments_02``` for example.

Reply:
It looks like commenting this code out doesn't change the behavior of `PrimaryConstructorTests.BaseArguments_02`.

Reply:
The test starts failing for me:
```
Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics.PrimaryConstructorTests.BaseArguments_02 [FAIL]
  System.InvalidOperationException : Assertion failed
```

Are you sure you are not running the record flavor of the test?

Reply:
I think it is this assert in ```BindPrimaryConstructorBody```:
```
                Debug.Assert(initializerBinder != null);
                initializer = initializerBinder.BindConstructorInitializer(baseWithArguments, diagnostics);
```

Reply:
Ok, I see what we were doing different. I was commenting out the code inside the override and was getting a failure, as expected. Removing the overload is not the right thing to do because it would visit way more things that we intend to visit.